rllib/env/multi_agent_env.py

```diff
@@ -408,37 +408,42 @@ def __init__(self, config=None):
             else:
                 self.agents = [env_name_or_creator(config) for _ in range(num)]
             self.dones = set()
-            self.observation_space = self.agents[0].observation_space
-            self.action_space = self.agents[0].action_space
+            self._spaces_in_preferred_format = True
+            self.observation_space = gym.spaces.Dict(
```

Comment:
can I make a quick coding suggestion, as written, the keys of self.observation_space and self.action_space are only correct if self._agent_ids stays set(range(num)).
a safer way to do this maybe to create self._agent_ids first, then:

```
self._agent_ids = set(range(num))
self.observation_space = gym.spaces.Dict(
    {i: a.observation_space for i, a in zip(self._agent_ids, self.agents)}
)
self.action_space = gym.spaces.Dict(
    {i: a.action_space for i, a in zip(self._agent_ids, self.agents)}
)
```
definite a nit.